src/kOS/Communication/CommNetConnectivityManager.cs

```diff
@@ -105,15 +105,15 @@ public bool HasConnection(Vessel vessel1, Vessel vessel2)
                 return true;
 
             // We next need to query the network to find a connection between the two vessels.
-            // I found no exposed method for accessing chached paths directly, other than the
+            // I found no exposed method for accessing cached paths directly, other than the
             // control path.
 
             // WARNING: In stock this will only work for vessels with a relay antenna installed.
             // This is a limitation put in place to improve performance in the stock game, and
             // there isn't a very good way around it.
             var net = CommNetNetwork.Instance.CommNet;
             tempPath = new CommPath();
-            return net.FindPath(vessel1.Connection.Comm, tempPath, vessel2.Connection.Comm) || net.FindPath(vessel2.Connection.Comm, tempPath, vessel1.Connection.Comm);
+            return vessel1 == vessel2 || net.FindPath(vessel1.Connection.Comm, tempPath, vessel2.Connection.Comm) || net.FindPath(vessel2.Connection.Comm, tempPath, vessel1.Connection.Comm);
```

Comment:
Should we be using  a raw reference-equals check here?  We seem to be using ``vessel.id`` for this sort of check in a lot of places elsewhere in the code.

Granted, it would be really weird for there to be two instances of ``Vessel`` that are really the same vessel, so it probably doesn't matter, but I wouldn't put it past KSP to have something that weird happening.